redbot/cogs/mod/events.py

```diff
@@ -19,20 +20,30 @@ class Events(MixinMeta):
 
     async def check_duplicates(self, message):
         guild = message.guild
-        author = message.author
+        author = message.author.id
+        channel = message.channel
 
-        guild_cache = self.cache.get(guild.id, None)
-        if guild_cache is None:
-            repeats = await self.config.guild(guild).delete_repeats()
+        repeats = await self.config.guild(guild).delete_repeats()
+        repeats_channels = await self.config.guild(guild).delete_repeats_channels.all()
+        if str(channel.id) in repeats_channels:
+            if repeats_channels[str(channel.id)] == -1:
+                return False
+        else:
             if repeats == -1:
                 return False
-            guild_cache = self.cache[guild.id] = defaultdict(lambda: deque(maxlen=repeats))
 
+        guild_cache = self.cache.get(guild.id, None)
+        if guild_cache is None:
+
+            # Cache is Map (channel -> Map (author -> messages))
+            guild_cache = self.cache[guild.id] = await create_new_cache(self.config, guild)
         if not message.content:
             return False
 
-        guild_cache[author].append(message.content)
-        msgs = guild_cache[author]
+        guild_cache[str(channel.id)][author].append(message.content)
```

Comment:
I honestly cannot tell if it's safe to do this with all the different conditions for how `guild_cache` can be set up and how this line can be reached.

Reply:
Could I have some more informations about this? Because I didn't change much about this part of the code except for the dict because I needed a dictionary per channel.

Reply:
https://github.com/Cog-Creators/Red-DiscordBot/blob/fe92cd88a2d22fe7935b356e8e914d2182de003d/redbot/cogs/mod/utils.py#L20-L24

`guild_cache` *may* be a `defaultdict[defaultdict[int]]`, though it is not guaranteed to be one. Thus, I can't tell if blindly accessing `guild_cache[str(channel.id)][author]` is safe to do from reading the code. This kind of complexity is a bad code smell, so the way this var is created and used should be rewritten to be clearer.

Reply:
Ok I understand and I already made the changes. It is possible to create a deque with maxlen = 0. Thus these checks aren't useful anymore and have been deleted

Reply:
Just letting you know in case, you have not pushed any changes to this PR since my review. If you're still working, and you mean that you made the changes locally, that's fine. I just can't see any changes so I can only base what I say on what I can see.

Reply:
Sorry for the misunderstanding, yes I changed it locally I still have to update my code with these reviews but I will push it in a few days